build_runner/lib/src/generate/input_matcher.dart

```diff
@@ -61,14 +63,35 @@ class _GlobInputMatcher implements InputMatcher {
   bool operator ==(Object other) =>
       identical(this, other) ||
       (other is _GlobInputMatcher &&
-          _deepEquals.equals(_patterns(include), _patterns(exclude)) &&
+          _deepEquals.equals(_patterns(include), _patterns(other.include)) &&
           _deepEquals.equals(_patterns(exclude), _patterns(other.exclude)));
 
   @override
   int get hashCode =>
       _deepEquals.hash([_patterns(include), _patterns(exclude)]);
 }
 
+class _MultiMatcher implements InputMatcher {
```

Comment:
nit: add a comment indicating this takes the intersection of the delegate matchers

Reply:
Added as a comment on the factor constructor above.